src/librustc_mir/const_eval.rs

```diff
@@ -24,7 +24,8 @@ use crate::interpret::{self,
     RawConst, ConstValue,
     InterpResult, InterpErrorInfo, InterpError, GlobalId, InterpCx, StackPopCleanup,
     Allocation, AllocId, MemoryKind,
-    snapshot, RefTracking, intern_const_alloc_recursive,
+    snapshot, RefTracking, intern_const_alloc_recursive, UnsupportedInfo::*,
+    InvalidProgramInfo::*,
```

Comment:
These enums contain tons of variants. They should only be glob-imported very locally around a `match`. Please don't glob-import enums for an entire file.

Reply:
Is there a performance difference if you glob-import enums for an entire file VS glob importing around a match?

Reply:
I don't know anything much about compilers, but I thought imports are handled at compile time.

Reply:
I think they are talking that glob-importing may bloat namespace and causing
import conflict in the future.

Reply:
I don't know if there is a performance difference. What I am worried about is namespace pollution. I don't want all those random names from the enum float around the namespace in that entire file. That's IMO bad style, it is not the usual Rust style, and it is certainly not rustc style.

Reply:
In particular for our case here, whether something is in `UnsupportedOp` or `UndefinedBehavior` makes a *huge* difference, and when raising an error we have to be sure we are raising the right kind of error.

Reply:
So, in this specific file `UnsupportedInfo` is used in more than one function. So do I use just import `UnsupportedInfo` at a global level OR import `UnsupportedInfo::*` in those functions?

Reply:
The preferred way is to import `UnsupportedInfo` globally. Then, if one function needs a *lot* of variants (like the `Debug` impls do), you do `use UnsupportedInfo::*;` *inside* that function.